setup.py

```diff
@@ -2,18 +2,30 @@
 from codecs import open
 from os import path
 
-
 # from Michael Hoffman's http://www.ebi.ac.uk/~hoffman/software/sunflower/
 class NumpyExtension(Extension):
 
     def __init__(self, *args, **kwargs):
+        Extension.__init__(self, *args, **kwargs)
+
+        self._include_dirs = self.include_dirs
+        del self.include_dirs  # restore overwritten property
+
+    # warning: Extension is a classic class so it's not really read-only
+
+    def get_include_dirs(self):
         from numpy import get_include
-        from numpy.distutils.misc_util import get_info
-        kwargs.update(get_info('npymath'))
```

Comment:
I saw these lines, but isn't it useless to update the kwargs after the Extension has been initialized with the original kwargs? Or the kwargs passed by reference and are consumed afterwards?